### DISCUSSION TITLE
Wagmi Hooks on mobile to open wallet app for interaction

### DISCUSSION THREAD

**Discussion by lifeofpavs** [wevm/wagmi#1128] [Q&A]
Hi guys! I have been working with wagmi lately and would love to bring its capabilities to mobile web apps. 
Wallet connection works good on `Metamask` on my device  it automatically opens the app to connect) but other actions like `useSignMessage` or `useContract` do not open the app although they trigger the action. 

When I go `Metamask` after triggering the action on the browser, there is a popup asking for confirmation, but I would like to redirect the user to the wallet instead of having them opening the app manually and then going back

Any idea why this could be happening or any workaround? 
Thanks a lot! 

PS: I am testing it on my phone (iOS 16) with Safari and Chrome

**Comment by adrianotadao** [comment]
hey @pabloadell this is exactly what I need. Did you do something about it?

**Reply by lifeofpavs** [reply]
Hey @adrianotadao! Yes, I ended up using a deep link in order to use the app --> https://metamask.github.io/metamask-deeplinks/. hope this helps you!

**Reply by adrianotadao** [reply]
cool, I will try the same. Thank you!

**Reply by adrianotadao** [reply]
not sure if there is a better solution but I ended up doing as following:

```javascript

const { signMessage } = useSignMessage();
signMessage({ message });

Linking.openURL('https://metamask.app.link/wc'); // this is on native
```

this will only open the wallet as I needed.
